monitor/internal/pod/delete_controller.go

```diff
@@ -114,19 +126,42 @@ func deleteControllerProcessItem(backgroundCtx context.Context, c client.Client,
 
 	// the edge case: a pod with the same namespaced name came up and we have missed a delete event
 	// this means that this pod belongs to a different PU and must live, therefore we try to delete the old one
-	if nativeID != string(pod.GetUID()) {
-		zap.L().Error("Pod does not have expected native ID, we must have missed an event and the same pod was recreated. Trying to destroy PU", zap.String("puID", nativeID), zap.String("namespacedName", req.String()), zap.String("podUID", string(pod.GetUID())))
-		if err := pc.Policy.HandlePUEvent(
-			ctx,
-			nativeID,
-			common.EventDestroy,
-			policy.NewPURuntimeWithDefaults(),
-		); err != nil {
-			// we don't really care, we just warn
-			zap.L().Warn("failed to handle destroy event", zap.String("puID", nativeID), zap.String("namespacedName", req.String()), zap.Error(err))
+	//if nativeID != string(pod.GetUID()) {
+	// Now destroy the PU only if the following
+	// 1. if the namesapce-name match && pod UID match
+	// 2. SandboxID differ.
+	podNamespaceName, err := client.ObjectKeyFromObject(pod)
```

Comment:
I usually just construct it from a new struct to avoid having to deal with the error that seems pointless :)